cpp/oneapi/dal/algo/svm/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -14,16 +14,109 @@
+                                     const table& data) {
+    auto queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const int64_t row_count             = data.get_row_count();
+    const int64_t column_count          = data.get_column_count();
+    const int64_t support_vectors_count = trained_model.get_support_vector_count();
+
+    // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
+    auto arr_data = row_accessor<const Float>{ data }.pull(queue);
+    auto arr_support_vectors =
+        row_accessor<const Float>{ trained_model.get_support_vectors() }.pull(queue);
+    auto arr_coefficients =
+        row_accessor<const Float>{ trained_model.get_coefficients() }.pull(queue);
+
+    const auto daal_data =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_data, row_count, column_count);
+    const auto daal_support_vectors =
+        interop::convert_to_daal_sycl_homogen_table(queue,
+                                                    arr_support_vectors,
+                                                    support_vectors_count,
+                                                    column_count);
+    const auto daal_coefficients =
+        interop::convert_to_daal_sycl_homogen_table(queue,
+                                                    arr_coefficients,
+                                                    support_vectors_count,
+                                                    1);
+
+    auto daal_model = daal_model_builder{}
+                          .set_support_vectors(daal_support_vectors)
+                          .set_coefficients(daal_coefficients)
+                          .set_bias(trained_model.get_bias());
+
+    const auto daal_kernel = desc.get_kernel_impl()->get_impl()->get_daal_kernel_function();
+    daal_svm::Parameter daal_parameter(daal_kernel);
+
+    auto arr_decision_function = array<Float>::empty(queue, row_count * 1);
+    const auto daal_decision_function =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_decision_function, row_count, 1);
+
+    daal_svm_predict_kernel_t<Float>().compute(daal_data,
+                                               &daal_model,
+                                               *daal_decision_function,
+                                               &daal_parameter);
+
+    auto arr_label = array<Float>::empty(queue, row_count * 1);
+    auto event     = queue.submit([&](sycl::handler& cgh) {
+        arr_label.need_mutable_data(queue);
+        arr_decision_function.need_mutable_data(queue);
+        auto labels            = arr_label.get_mutable_data();
+        auto decision_function = arr_decision_function.get_mutable_data();
+        cgh.parallel_for(sycl::range<1>(arr_label.get_count()), [=](sycl::id<1> idx) {
+            labels[idx[0]] = decision_function[idx[0]] >= 0 ? Float(1.0) : Float(-1.0);
```

Comment:
You don't need mutable data for decision function, because it's read only.

Reply:
Fixed